src/audio/pipeline/pipeline-stream.c

```diff
@@ -154,7 +155,7 @@ static int pipeline_comp_list(struct comp_dev *current,
 	bool is_same_sched = pipeline_is_same_sched_comp(current->pipeline,
 							 ppl_data->start->pipeline);
 
-	if (!is_single_ppl && !is_same_sched) {
+	if (!is_single_ppl && (!is_same_sched || IPC4_MOD_ID(current->ipc_config.id))) {
```

Comment:
getting complex, best to have a comment to explain.

Reply:
Done